src/components/webviews/jsInteractions/jsLogInterception.ts

```diff
@@ -23,19 +23,30 @@ export const jsLogInterception = `
   }
 `
 
-export const tryConsole = (payload, logger, logId) => {
+export const tryConsole = (
+  payload: { nativeEvent: { data: string } } | undefined,
+  logger: MiniLogger | undefined,
+  logId: string | undefined
+): void => {
   try {
+    if (!payload || !logger || !logId) return console.error('no payload')
+
     const { data: rawData } = payload.nativeEvent
 
-    const dataPayload = JSON.parse(rawData)
+    const dataPayload = JSON.parse(rawData) as {
+      type: string
+      data?: { type: string; log: string[] }
+    }

```

Comment:
nit: would be more correct to remove this `as` and to convert the next `if` to a typeguard

Reply:
the issue is JSON methods all return any, have to cast them